samtranslator/model/stepfunctions/generators.py

```diff
@@ -111,6 +111,7 @@ def to_cloudformation(self):
             )
         elif self.definition:
             processed_definition = deepcopy(self.definition)
+            processed_definition = self.intrinsics_resolver.resolve_parameter_refs(processed_definition)
```

Comment:
Couldn't we simply leave the `Ref` or reference the substitution variable as a non-string?

I wouldn't expect to see `"TimeoutSeconds": 500` in the transformed template.

Wouldn't this work?

`\"TimeoutSeconds\": ${definition_substitution_3}`

instead of 

`\"TimeoutSeconds\": \"${definition_substitution_3}\"`

I guess this would require SAM to know which properties are strings and which aren't?

Reply:
@jfuss It seems like this fix is stalled. I don't know the codebase at all, but I wanted to see if I could at least get the discussion going again to see if you and @vaib-amz could sort of if the approach being taken works.

I really hope I don't have to continue using JSON for some of my state machines (or render them using jinja)! Thank you all for your hard work.